extensions/wikia/ImageReview/StatsDataGetter.class.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+class StatsDataGetter extends WikiaModel {
+	public function getStatsData( $startYear, $startMonth, $startDay, $endYear, $endMonth, $endDay ) {
+
+		$startDate = $startYear . '-' . $startMonth . '-' . $startDay . ' 00:00:00';
+		$endDate = $endYear . '-' . $endMonth . '-' . $endDay . ' 23:59:59';
+
+		$summary = [
+			'all' => 0,
+			ImageStates::APPROVED 	=> 0,
+			ImageStates::REJECTED 	=> 0,
+			ImageStates::QUESTIONABLE => 0,
+			'avg' => 0,
+		];
+		$data = [];
+		$total = 0;
+
+		$dbr = $this->getDatawareDB( DB_SLAVE );
+		$reviewers = $this->getReviewersForStats();
+
+		$count_users = count( $reviewers );
+		if ( $count_users > 0 ) {
+			foreach ( $reviewers as $reviewer ) {
+				# user
+				$user = User::newFromId( $reviewer );
+				if ( !is_object( $user ) ) {
+					// invalid user id?
+					continue;
+				}
+
+				$query = [];
+				foreach ( array_keys( $summary ) as $review_state ) {
+					# union query: mysql explain: Using where; Using index and max 150k rows
+					$query[] = $dbr->selectSQLText(
+						[ 'image_review_stats' ],
+						[ 'review_state', 'count(*) as cnt' ],
+						[
+							"review_state"	=> $review_state,
+							"reviewer_id"	=> $reviewer,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "reviewer_id" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)